poetry/console/config/application_config.py

```diff
@@ -104,6 +115,35 @@ def set_env(self, event, event_name, _):  # type: (PreHandleEvent, str, Any) ->
 
         command.set_env(env)
 
+    def set_installer(
+        self, event, event_name, _
+    ):  # type: (PreHandleEvent, str, Any) -> None
+        command = event.command.config.handler  # type: WithInstaller
+        if not isinstance(command, WithInstaller):
+            return
+
+        if command.installer is not None:
+            return
+
+        if not isinstance(command, EnvCommand):
+            raise RuntimeError(
+                "A command that needs an installer must also inherit from EnvCommand"
```

Comment:
Since this is the case, wouldn't it be better if we have a new class `InstallerCommand(EnvCommand, WithInstaller)`? Additionally, can we consider removing the `WithInstaller` mixin since it might be better to have a child class that has an `installer` attribute. The mixin simply is adding an unnecessary abstraction here at the moment.

Furthermore, we could also consider doing away with this method in favour of `installer` being initialised within `InstallerCommand.__init__` as we should have all the values required at that point.

Reply:
Having an `InstallerCommand` class is actually a good idea.

When you say

> we could also consider doing away with this method

which method are you referring to.

Reply:
The `set_installer` methods. Right now we seem to use this in order to defer the installer creation until the required variables are populated. With the introduction of an `InstallerCommand` class we can make this more seamless.

Reply:
We can't instantiate the installer in `__init__` since at this point (the console application creation) we don't have all the information.